### ISSUE TITLE
Service tendrl-monitoring-integration fails during tendrl-ansible driven installation

### ISSUE THREAD

**Issue by mbukatov** [Tendrl/monitoring-integration#33]
## Description

When Tendrl is installed via [tendrl-ansible](https://github.com/Tendrl/tendrl-ansible), service `tendrl-monitoring-integration` fails early during startup so that after ansible reports the process as finished without any errors, service `tendrl-monitoring-integration` is not running and Tendrl dashboards are not loaded in Grafana.

## Version

I'm using packages from [tendrl/tendrl COPR](https://copr.fedorainfracloud.org/coprs/tendrl/tendrl/), which contains latest snapshot builds from master branch, when it contained builds for the [2nd milestone](https://github.com/Tendrl/specifications/milestone/3).

```
# rpm -qa tendrl-*
tendrl-monitoring-integration-1.0.0-20170823T072403.97cbe3c.noarch
tendrl-commons-1.5.0-20170823T172713.25e15fd.noarch
tendrl-api-1.5.0-20170823T134716.16b4bac.noarch
tendrl-node-agent-1.5.0-20170823T144615.8b070cc.noarch
tendrl-ui-1.5.0-20170823T145023.0ebf53f.noarch
tendrl-api-httpd-1.5.0-20170823T134716.16b4bac.noarch
# rpm -qa | egrep '(etcd|grafana|graphite|puma)'
graphite-web-0.9.16-1.el7.noarch
etcd-3.1.9-1.el7.x86_64
grafana-4.4.3-1.x86_64
rubygem-puma-3.6.0-1.el7.centos.x86_64
rubygem-etcd-0.3.0-1.el7.centos.noarch
python-etcd-0.4.5-1.noarch
```

Branch of pull request [Update tendrl-ansible for Tendrl Milestone](https://github.com/Tendrl/tendrl-ansible/pull/24) for tendrl-ansible was used.

## Reproducer

1. Prepare few storage machines and one server machine (for tendrl itself)
2. Install Tendrl via tendrl-ansible
3. Connect to the Tendrl Server machine and check if tendrl-integration-monitoring service is up

This is not reproducible during manual installation, you need to use ansible.

## Actual Results

The service is down, logs shows several unsuccessful attempts to start it:

```
[root@mbukatov-usm1-server ~]# journalctl -u tendrl-monitoring-integration
-- Logs begin at Wed 2017-08-23 12:44:36 UTC, end at Wed 2017-08-23 15:49:56 UTC. --
Aug 23 13:00:09 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:09 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:09 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: Creating namespace.monitoring from source tendrl.monitoring_integrationnamespace.monitoring created!Finding objects in namespace.monitoring.objectsRegistering object namespace.monitoring.objects.ConfigFinding atoms
Aug 23 13:00:09 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: toms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in 
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: e.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering ob
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: ts in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextLoad definitions (.yml) for namespace.tendrl.objects.TendrlContextTraceback (most 
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: File "/usr/bin/tendrl-monitoring-integration", line 9, in <module>
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: load_entry_point('tendrl-monitoring-integration==1.0.0', 'console_scripts', 'tendrl-monitoring-integration')()
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 143, in main
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: monitoring_integration_manager.start()
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 40, in start
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: _upload_default_dashboards()
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 62, in _upload_default_dashboards
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: dashboards = dashboard.get_all_dashboards()
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/grafana/dashboard.py", line 60, in get_all_dashboards
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: raise exceptions.ConnectionFailedException
Aug 23 13:00:11 mbukatov-usm1-server.example.com tendrl-monitoring-integration[15775]: tendrl.monitoring_integration.grafana.exceptions.ConnectionFailedException

Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service: main process exited, code=exited, status=1/FAILURE
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-monitoring-integration.service entered failed state.
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service failed.
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service holdoff time over, scheduling restart.
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Stopping Monitoring Integration...
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:11 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: Stopping Monitoring Integration...
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: Creating namespace.monitoring from source tendrl.monitoring_integrationnamespace.monitoring created!Finding objects in namespace.monitoring.objectsRegistering object namespace.monitoring.objects.ConfigFinding atoms
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: toms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in 
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: e.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering ob
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: ts in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextLoad definitions (.yml) for namespace.tendrl.objects.TendrlContextTraceback (most 
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: File "/usr/bin/tendrl-monitoring-integration", line 9, in <module>
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: load_entry_point('tendrl-monitoring-integration==1.0.0', 'console_scripts', 'tendrl-monitoring-integration')()
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 143, in main
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: monitoring_integration_manager.start()
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 40, in start
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: _upload_default_dashboards()
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 62, in _upload_default_dashboards
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: dashboards = dashboard.get_all_dashboards()
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/grafana/dashboard.py", line 60, in get_all_dashboards
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: raise exceptions.ConnectionFailedException
Aug 23 13:00:12 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16139]: tendrl.monitoring_integration.grafana.exceptions.ConnectionFailedException
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service: main process exited, code=exited, status=1/FAILURE
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-monitoring-integration.service entered failed state.
Aug 23 13:00:12 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service failed.
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service holdoff time over, scheduling restart.
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: Creating namespace.monitoring from source tendrl.monitoring_integrationnamespace.monitoring created!Finding objects in namespace.monitoring.objectsRegistering object namespace.monitoring.objects.ConfigFinding atoms
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: toms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in 
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: e.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering ob
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: ts in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextLoad definitions (.yml) for namespace.tendrl.objects.TendrlContextTraceback (most 
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: File "/usr/bin/tendrl-monitoring-integration", line 9, in <module>
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: load_entry_point('tendrl-monitoring-integration==1.0.0', 'console_scripts', 'tendrl-monitoring-integration')()
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 143, in main
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: monitoring_integration_manager.start()
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 40, in start
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: _upload_default_dashboards()
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 62, in _upload_default_dashboards
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: dashboards = dashboard.get_all_dashboards()
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/grafana/dashboard.py", line 60, in get_all_dashboards
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: raise exceptions.ConnectionFailedException
Aug 23 13:00:13 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16160]: tendrl.monitoring_integration.grafana.exceptions.ConnectionFailedException
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service: main process exited, code=exited, status=1/FAILURE
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-monitoring-integration.service entered failed state.
Aug 23 13:00:13 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service failed.
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service holdoff time over, scheduling restart.
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: Started Monitoring Integration.
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: Starting Monitoring Integration...
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: Creating namespace.monitoring from source tendrl.monitoring_integrationnamespace.monitoring created!Finding objects in namespace.monitoring.objectsRegistering object namespace.monitoring.objects.ConfigFinding atoms
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: toms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in 
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: e.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering ob
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: ts in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextLoad definitions (.yml) for namespace.tendrl.objects.TendrlContextTraceback (most 
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: File "/usr/bin/tendrl-monitoring-integration", line 9, in <module>
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: load_entry_point('tendrl-monitoring-integration==1.0.0', 'console_scripts', 'tendrl-monitoring-integration')()
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 143, in main
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: monitoring_integration_manager.start()
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 40, in start
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: _upload_default_dashboards()
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 62, in _upload_default_dashboards
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: dashboards = dashboard.get_all_dashboards()
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/grafana/dashboard.py", line 60, in get_all_dashboards
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: raise exceptions.ConnectionFailedException
Aug 23 13:00:14 mbukatov-usm1-server.example.com tendrl-monitoring-integration[16172]: tendrl.monitoring_integration.grafana.exceptions.ConnectionFailedException
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service: main process exited, code=exited, status=1/FAILURE
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-monitoring-integration.service entered failed state.
Aug 23 13:00:14 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service failed.
Aug 23 13:00:15 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service holdoff time over, scheduling restart.
Aug 23 13:00:15 mbukatov-usm1-server.example.com systemd[1]: start request repeated too quickly for tendrl-monitoring-integration.service
Aug 23 13:00:15 mbukatov-usm1-server.example.com systemd[1]: Failed to start Monitoring Integration.
Aug 23 13:00:15 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-monitoring-integration.service entered failed state.
Aug 23 13:00:15 mbukatov-usm1-server.example.com systemd[1]: tendrl-monitoring-integration.service failed.
```

Here is the exception extracted from the logs:

```
Creating namespace.monitoring from source tendrl.monitoring_integrationnamespace.monitoring created!Finding objects in namespace.monitoring.objectsRegistering object namespace.monitoring.objects.ConfigFinding atoms
toms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in 
e.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering ob
ts in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextLoad definitions (.yml) for namespace.tendrl.objects.TendrlContextTraceback (most 
File "/usr/bin/tendrl-monitoring-integration", line 9, in <module>
load_entry_point('tendrl-monitoring-integration==1.0.0', 'console_scripts', 'tendrl-monitoring-integration')()
File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 143, in main
monitoring_integration_manager.start()
File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 40, in start
_upload_default_dashboards()
File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/manager/__init__.py", line 62, in _upload_default_dashboards
dashboards = dashboard.get_all_dashboards()
File "/usr/lib/python2.7/site-packages/tendrl/monitoring_integration/grafana/dashboard.py", line 60, in get_all_dashboards
raise exceptions.ConnectionFailedException
tendrl.monitoring_integration.grafana.exceptions.ConnectionFailedException
```

Which suggests that the service failed to connect to Grafana.

## Expected Results

The daemon handles the failure and tries to establish the connection again few times (like 3 or 5 times) with some delay between the retries, without ending with an exception. Or maybe it could report the problem in the logs, keep running and retry again after some time.

## Additional Details

The problem here seems to be the same to the issue with `tendrl-monitoring-integration` script from the previous release: when the script was executed via ansible, grafana (or whatever the component monitoring-integration tries to speak with) wasn't yet ready to respond and so the script failed.

I addressed the problem in tendrl-ansible by adding `retries: 5` and `delay: 5` to the task running the script, based on discussion with @rishubhjain. But since the script were turned into a service, it needs to handle this itself.

## Workaround

When you rerun the playbook or start the service again manually, it works just fine (as the Grafana component is already able to respond to tendrl-monitorin-integration queries).

**Comment by nthomas-redhat** [comment]
Looks like a timing issue. I haven't seen this on our testbeds

**Comment by rishubhjain** [comment]
@mbukatov A solution to this issue has been merged,https://github.com/Tendrl/monitoring-integration/issues/61
 could you test out and inform whether this issue is still reproducible?

**Comment by mbukatov** [comment]
I retried with snapshot builds from master (via `tendrl_copr_repo: "master"` variable of tendrl-ansible), which installed `tendrl-monitoring-integration-1.5.1-20170908T124712.dadabad.noarch` package, and the issue is still there (tendrl-monitoring-integration service was down when the playbook finished, but manual start of the service worked).

@rishubhjain could you check if the fix is already in the build I used?

**Comment by mbukatov** [comment]
I retried with `tendrl-monitoring-integration-1.5.1-20170913T163914.7eb80a2.noarch` and the problem persists.

**Comment by TimothyAsirJeyasing** [comment]
Seems like some connectivity issue.
Add after clause in the systemd service configuration
After= network.target multi-user.target

**Comment by mbukatov** [comment]
Hmm, but the unit file already contains:

```
[Unit]
Description= Monitoring Integration
After=grafana-server.service
Requires=tendrl-node-agent.service grafana-server.service
```

would that make any difference?

**Comment by rishubhjain** [comment]
@mbukatov I guess to handle this issue we will have to place some workarounds.
If possible can you handle this issue via ansible?

One functionality provided by ansible can be helpful to solve this issue
http://docs.ansible.com/ansible/latest/wait_for_module.html

**Comment by mbukatov** [comment]
@rishubhjain such workaround would work only if we know what the root cause is exactly and at the same time we are sure that the problem doesn't occur in any other case. Based on my understanding of the problem so far, this is not the case.

**Comment by mbukatov** [comment]
The issue seems resolved since `tendrl-monitoring-integration-1.5.1-20170915T172525.660d31a.noarch`, the service is up and running, there is no evidence of a crash in tendrl-monitoring-integration logs.
